src/main/java/teammates/ui/controller/ImageUploadAction.java

```diff
@@ -0,0 +1,99 @@
+package teammates.ui.controller;
+
+import java.util.List;
+import java.util.Map;
+
+import teammates.common.util.Config;
+import teammates.common.util.Const;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreFailureException;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+
+public class ImageUploadAction extends Action {
+
+    FileUploadPageData data;
+
+    @Override
+    protected ActionResult execute() {
+        data = new FileUploadPageData(account);
+        BlobInfo blobInfo = extractImageKey(Const.ParamsNames.IMAGE_TO_UPLOAD);
+
+        if (blobInfo == null) {
+            data.isFileUploaded = false;
+            data.fileSrcUrl = null;
+            log.warning("Image Upload Failed");
+            statusToAdmin = "Image Upload Failed";
+
+            return createAjaxResult(data);
+        }
+
+        BlobKey blobKey = blobInfo.getBlobKey();
+
+        data.isFileUploaded = true;
+        data.fileSrcUrl = Const.ActionURIs.PUBLIC_IMAGE_SERVE + "?blob-key=" + blobKey.getKeyString();
+        String absoluteFileSrcUrl = Config.getAppUrl(data.fileSrcUrl).toAbsoluteString();
```

Comment:
Since we are refactoring this method anyway, let's rewrite this part as well:
```java
AppUrl fileSrcUrl = Config.getAppUrl(Const.ActionURIs.PUBLIC_IMAGE_SERVE).with...(...);
String absoluteFileSrcUrl = fileSrcUrl.toAbsoluteString();
data.fileSrcUrl = fileSrcUrl.toString();
```